Modules/CodeTrek/Services/CodeTrekService.php

```diff
@@ -123,4 +125,23 @@ public function storeCodeTrekApplicantFeedback($data)
         $feedback->posted_on = today();
         $feedback->save();
     }
+
+    public function getCandidateFeedbacks($data)
+    {
+        $feedbacks = CodeTrekCandidateFeedback::where('candidate_id', $data)->get();
+        $candidateFeedbacks = [];
+        foreach ($feedbacks as $feedback) {
+            $candidateFeedback = new \stdClass();
+            $candidateFeedback->id = $feedback['id'];
+            $candidateFeedback->posted_by = User::select('name')->where('id', $feedback['posted_by'])->first()->name;
+            $candidateFeedback->posted_on = $feedback['posted_on'];
+            $candidateFeedback->feedback_category = CodeTrekFeedbackCategories::select('category_name')->where('id', $feedback['category_id'])->first()->category_name;
```

Comment:
I think this code is not properly applied the single responsibility principle. you can make the 2 private functions for the `User` name and `FeedbackCategories`. and then you can write like this here. And I think we can be optimized this code.

```suggestion
            $candidateFeedback->id = $feedback['id'];
            $candidateFeedback->posted_by =  $this->getPostedByUserName($feedback['posted_by']);
            $candidateFeedback->posted_on = $feedback['posted_on'];
            $candidateFeedback->feedback_category = $this->getFeedbackCategoryName($feedback['category_id']);
```